k8srm-prototype/pkg/api/capacity_types.go

```diff
@@ -40,9 +40,17 @@ type DevicePoolSpec struct {
 	// +optional
 	Attributes []Attribute `json:"attributes,omitempty"`
 
-	// DeviceCount contains the total number of devices in the pool.
+	// Resources are pooled resources that are shared by all devices in the
+	// pool. This is typically used when representing a partitionable
+	// device, and need not be populated otherwise.
+	//
+	// +optional
+	Resources []ResourceCapacity `json:"resources,omitempty"`
```

Comment:
Also, It feels artificially limiting to only support `ResourceCapacity` here. Many resources can be better represented as a set and *forcing* one to always use a resource.Quantity under the hood seems unnecessary. Can we wrap this in a one-of-many type so that this can be extended in the future with other types (e.g. my IntRange type).

Reply:
Yeah, going for fewer concepts and types. We'll discuss.

Reply:
In my earlier model, I had integer, quantity, blocks quantity, and access mode all modeled *under* resource capacity. IntSlice would have fit in nicely there. We could add an indirection here to enable that again in the future but it does make understanding it harder. But just trying to cut back on complexity.